pkg/service/providercacher/simpleprovidercacher.go

```diff
@@ -18,20 +20,40 @@ func NewSimpleProviderCacher(providerStore types.ProviderStore) ProviderCacher {
 
 func (s *simpleProviderCacher) CacheProviderForIndexRecords(ctx context.Context, provider model.ProviderResult, index blobindex.ShardedDagIndexView) (uint64, error) {
 	written := uint64(0)
+
+	// Prioritize the root
+	rootDigest := link.ToCID(index.Content()).Hash()
```

Comment:
essentially, adding the root CID to the cache first in case something fails later when adding the shards, right?